In [2]:
import numpy as np
import pandas as pd
import matplotlib
import seaborn
import re

# Importing data

In [5]:
!pip install ucimlrepo

In [6]:
from ucimlrepo import fetch_ucirepo

In [7]:
# fetch dataset 
raw_dataset = fetch_ucirepo(id=2) 

In [8]:
print(type(raw_dataset))

<class 'ucimlrepo.dotdict.dotdict'>


This object is an instance of "dotdict" which is a custom dictionary-like structure from UCIMRepo library. Unlike the normal dictionary where indexing is used to access values, dot notation is used in this type of object.

In [10]:
# Import training features "X" and target column "y"
X = raw_dataset.data.features 
y = raw_dataset.data.targets

In [11]:
# Check type of the datasets
print(type(X))
print(type(y))

<class 'pandas.core.frame.DataFrame'>
<class 'pandas.core.frame.DataFrame'>


Both are pandas dataframe type.

In [12]:
# Evaluate training features
X.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country  
0          2174             0              40  United-States  
1             0             0              13  United-States  
2             0             0              40  United-States  
3             0             0              40  United-States  
4             0             0              40           Cuba

In [14]:
# Basic statistics of the training dataset
X.describe(include="all")

age workclass        fnlwgt education  education-num  \
count   48842.000000     47879  4.884200e+04     48842   48842.000000   
unique           NaN         9           NaN        16            NaN   
top              NaN   Private           NaN   HS-grad            NaN   
freq             NaN     33906           NaN     15784            NaN   
mean       38.643585       NaN  1.896641e+05       NaN      10.078089   
std        13.710510       NaN  1.056040e+05       NaN       2.570973   
min        17.000000       NaN  1.228500e+04       NaN       1.000000   
25%        28.000000       NaN  1.175505e+05       NaN       9.000000   
50%        37.000000       NaN  1.781445e+05       NaN      10.000000   
75%        48.000000       NaN  2.376420e+05       NaN      12.000000   
max        90.000000       NaN  1.490400e+06       NaN      16.000000   

            marital-status      occupation relationship   race    sex  \
count                48842           47876        48842  48842  48842   
unique                   7              15            6      5      2   
top     Married-civ-spouse  Prof-specialty      Husband  White   Male   
freq                 22379            6172        19716  41762  32650   
mean                   NaN             NaN          NaN    NaN    NaN   
std                    NaN             NaN          NaN    NaN    NaN   
min                    NaN             NaN          NaN    NaN    NaN   
25%                    NaN             NaN          NaN    NaN    NaN   
50%                    NaN             NaN          NaN    NaN    NaN   
75%                    NaN             NaN          NaN    NaN    NaN   
max                    NaN             NaN          NaN    NaN    NaN   

        capital-gain  capital-loss  hours-per-week native-country  
count   48842.000000  48842.000000    48842.000000          48568  
unique           NaN           NaN             NaN             42  
top              NaN           NaN             NaN  United-States  
freq             NaN           NaN             NaN          43832  
mean     1079.067626     87.502314       40.422382            NaN  
std      7452.019058    403.004552       12.391444            NaN  
min         0.000000      0.000000        1.000000            NaN  
25%         0.000000      0.000000       40.000000            NaN  
50%         0.000000      0.000000       40.000000            NaN  
75%         0.000000      0.000000       45.000000            NaN  
max     99999.000000   4356.000000       99.000000            NaN

We can see that there are 48842 entries in the training dataset. There 14 unique feature columns that can be used for training.  
Above statistics show missing values each of the column if any.

In [15]:
# Evaluate target column
y.head()

income
0  <=50K
1  <=50K
2  <=50K
3  <=50K
4  <=50K

In [16]:
y.describe(include="all")

income
count   48842
unique      4
top     <=50K
freq    24720

As seen before, there are 48842 entries in the target column with 4 unique entries. 

In [21]:
# Merge X (features) and y (target) into a single DataFrame
dataframe = pd.concat([X, y], axis=1)

In [22]:
dataframe.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country income  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [23]:
# check of merged dataset
len(dataframe)

48842

Merged good since there are still 48842 entries.

In [26]:
# Looking at unique target columns
dataframe["income"].unique()

array(['<=50K', '>50K', '<=50K.', '>50K.'], dtype=object)

There are the four unique targets.

In [27]:
# Save to CSV file
dataframe.to_csv("dataframe.csv", index=False)